### PR TITLE
Fix vendor add bug

### LINKED ISSUE #109: Fix bug in vendor add command parser
The `vendor add` command parser only accepts `name` and `phone` arguments. It is supposed to accept the following parameters:
* `name`
* `phone`
* `email`
* `address`
* `tag`

Update `VendorAddCommandParser` to accept the remaining arguments, along with corresponding testcases.

### PR DESCRIPTION
**Pull request by p-xp**
Resolves #109.

`Vendor add` command does not accept user inputs for email and address fields.

This is contrary to the expecatations laid out in the User Guide.

Lets,
- Improve the implementation of the `vendor add` command so that email and address inputs will be tracked and stored by WedLog.
- Improve comprehensiveness of test cases such that similar bugs (i.e. inputs not being accepted) will be detected during automatic testing.
- Update display message for vendor add so that it reflects information for the two new fields.

### CODE REVIEW

File: src/test/java/wedlog/address/logic/commands/GuestAddCommandIntegrationTest.java [hunk 1]
```diff
@@ -26,6 +28,28 @@ public void setUp() {
...
+                expectedModel);
+
+        Guest insertedGuest = model.getFilteredGuestList().get(0);
+
+        // Check that all fields of the Guest have been stored properly in WedLog
+        assertEquals(insertedGuest.getName(), validGuest.getName());
+        assertEquals(insertedGuest.getPhone(), validGuest.getPhone());
+        assertEquals(insertedGuest.getEmail(), validGuest.getEmail());
+        assertEquals(insertedGuest.getAddress(), validGuest.getAddress());
+        assertEquals(insertedGuest.getTags(), validGuest.getTags());
```
Comment (samuelim01) [inline src/test/java/wedlog/address/logic/commands/GuestAddCommandIntegrationTest.java:50]: Is this line necessary? `assertCommandSuccess` already checks the `model` against the `expectedModel`, which checks between the `ModelManager` objects, which checks between the `AddressBook` objects, and eventually checks between the fields of each `Guest`.
```suggestion
```

File: src/test/java/wedlog/address/logic/commands/VendorAddCommandIntegrationTest.java [hunk 2]
```diff
@@ -37,6 +40,28 @@ public void execute_newVendor_success() {
...
                 expectedModel);
+
+        Vendor insertedVendor = model.getFilteredVendorList().get(0);
+
+        // Check that all fields of the Vendor have been stored properly in WedLog
+        assertEquals(insertedVendor.getName(), validVendor.getName());
+        assertEquals(insertedVendor.getPhone(), validVendor.getPhone());
+        assertEquals(insertedVendor.getEmail(), validVendor.getEmail());
+        assertEquals(insertedVendor.getAddress(), validVendor.getAddress());
+        assertEquals(insertedVendor.getTags(), validVendor.getTags());
```
Comment (samuelim01) [inline src/test/java/wedlog/address/logic/commands/VendorAddCommandIntegrationTest.java:51]: Is this line necessary? `assertCommandSuccess` already checks the `model` against the `expectedModel`, which checks between the `ModelManager` objects, which checks between the `AddressBook` objects, and eventually checks between the fields of each `Vendor`.
```suggestion
```

File: src/test/java/wedlog/address/logic/commands/GuestAddCommandIntegrationTest.java [hunk 3]
```diff
@@ -26,6 +28,28 @@ public void setUp() {
...
+    }
+
     @Test
     public void execute_newGuest_success() {
```
Comment (samuelim01) [inline src/test/java/wedlog/address/logic/commands/GuestAddCommandIntegrationTest.java:54]:
```suggestion
    public void execute_newGuestIntoPrefilledAddressBook_success() {
```

File: src/test/java/wedlog/address/logic/parser/VendorAddCommandParserTest.java
Comment (samuelim01) [inline src/test/java/wedlog/address/logic/parser/VendorAddCommandParserTest.java:1]: Perhaps `VendorAddCommandParserTest#parse_invalidValue_failure()` should include relevant tests for `email`, `address`, and `tag` too?

File: src/main/java/wedlog/address/model/person/DietaryRequirements.java [hunk 4]
```diff
@@ -65,6 +65,11 @@ public boolean equals(Object other) {
             return true;
         }
 
+        // Dietary Requirement where Status.NULL is equal to null object
+        if (other == null) {
+            return this.status.equals(Status.NULL);
+        }
+
```
Comment (samuelim01) [inline src/main/java/wedlog/address/model/person/DietaryRequirements.java:72]: Personally, I don't think this should be true. Do you have a particular use case in mind in writing this line?

**Review by samuelim01 — COMMENTED** [review]
LGTM, just a few nits

File: src/test/java/wedlog/address/logic/commands/GuestAddCommandIntegrationTest.java [hunk 5]
```diff
@@ -1,5 +1,6 @@
 package wedlog.address.logic.commands;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Comment (wasjoe1) [inline src/test/java/wedlog/address/logic/commands/GuestAddCommandIntegrationTest.java:3]:
```suggestion
```

File: src/test/java/wedlog/address/logic/commands/VendorAddCommandIntegrationTest.java [hunk 6]
```diff
@@ -1,5 +1,6 @@
 package wedlog.address.logic.commands;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Comment (wasjoe1) [inline src/test/java/wedlog/address/logic/commands/VendorAddCommandIntegrationTest.java:3]:
```suggestion
```

**Review by wasjoe1 — APPROVED** [review]
Some changes to be made in next iteration but LGTM for v1.2!

**Pull request merged**